Google.PowerShell/Common/GcloudAttributes.cs

```diff
@@ -117,4 +119,37 @@ public void SetConfigDefault(FieldInfo field, object instance)
             }
         }
     }
+    /// <summary>
+    /// This attribute allows an arry parameter to accept multiple types by replacing a target type with the 
+    /// value of a specified property. If the array element is not of the target type, it will not be changed.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// [ArrayPropertyTransform(typeof(Zone), nameof(Zone.Name))]
+    /// public string[] ZoneName { get; set; }
+    /// </code>
+    /// Transforms any Zone objects given to the ZoneName Parameter into zoneObject.Name
+    /// </example>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = true)]
+    public class ArrayPropertyTransformAttribute : ArgumentTransformationAttribute
+    {
+        private PropertyByTypeTransformationAttribute typeTransformationAttribute;
+
+        public ArrayPropertyTransformAttribute(Type typeToTransform, string property)
+        {
+            typeTransformationAttribute = new PropertyByTypeTransformationAttribute
+            {
+                Property = property,
+                TypeToTransform = typeToTransform
+            };
+        }
+
+        public override object Transform(EngineIntrinsics engineIntrinsics, object inputData)
```

Comment:
This snippet of code is pretty badass. You should have this snippet engraved on your tombstone or a tatoo or something.